### PR TITLE
feat: CompiledBinding StringFormat without curly braces

### LINKED ISSUE #11258: CompiledBinding StringFormat without curly braces
**Is your feature request related to a problem? Please describe.**
Simplify migration from WPF.

**Describe the solution you'd like**
Allow using binding StringFormat like `{Binding Value,StringFormat=c}` or `{Binding Value,StringFormat=c2}`

**Describe alternatives you've considered**

**Workaround**
`{Binding Value,StringFormat=\{0:c2\}}`

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Allow using binding StringFormat like `{Binding Value,StringFormat=c}` or `{Binding Value,StringFormat=c2}`

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Fixes #11258

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034553-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: tests/Avalonia.Markup.UnitTests/Data/BindingTests_Converters.cs [hunk 1]
```diff
@@ -52,6 +51,28 @@ public void StringFormatConverter_Should_Be_Used_When_Binding_Has_StringFormat()
...
+            {
+                var textBlock = new TextBlock
+                {
+                    DataContext = new Class1(),
+                };
+
+                var binding = new Binding(nameof(Class1.DecimalValue))
+                {
+                    StringFormat = "c2",
+                };
```
Comment (maxkatz6) [inline tests/Avalonia.Markup.UnitTests/Data/BindingTests_Converters.cs:67]: It's a reflection bindings test. 
If you want to test compiled bindings, would be easier to add XAML tests to the Avalonia.Markup.Xaml.UnitTests project.

Comment (workgroupengineering) [reply tests/Avalonia.Markup.UnitTests/Data/BindingTests_Converters.cs:67]: I've also seen that ReflectionBing doesn't work. I'll add it tomorrow

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034561-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by timunie** [comment]
this needs to be updated in the 11.0 docs as well, see: https://docs.avaloniaui.net/docs/data-binding/bindings#binding-stringformat

**Review by maxkatz6 — APPROVED** [review]
I think we don't need to update StringFormatMultiValueConverter, as it's expected to have at least two values normally.

**Pull request merged**